modelgym/gp_trainer.py

```diff
@@ -51,58 +41,18 @@ def crossval_optimize_params(self, model, cv_pairs, max_evals=None, verbose=True
-        if (isinstance(params, dict)):
+            res = super().crossval_fit_eval(model=model, cv_pairs=cv_pairs, n_estimators=n_estimators,
+                                            verbose=verbose)
+        elif (isinstance(params, dict)):
             params = model.preprocess_params(params)
+            res = super().crossval_fit_eval(model=model, cv_pairs=cv_pairs, params=params, n_estimators=n_estimators,
+                                            verbose=verbose)
+        else:
+            raise ValueError()
+        return res['loss']
 
-        mean_evals_results = []
-        std_evals_results = []
-
-        for dtrain, dtest in cv_pairs:
-            _dtrain = model.convert_to_dataset(dtrain.X, dtrain.y, dtrain.cat_cols)
-            _dtest = model.convert_to_dataset(dtest.X, dtest.y, dtest.cat_cols)
-            _, evals_result = model.fit(params, _dtrain, _dtest, n_estimators)
-            evals_results.append(evals_result)
-            mean_evals_results.append(np.mean(evals_result))
-            std_evals_results.append(np.std(evals_result))
-        best_n_estimators = np.argmin(mean_evals_results) + 1
-        eval_time = time.time() - start_time
-        loss = mean_evals_results[best_n_estimators - 1]
-
-        self.best_loss = min(self.best_loss, loss)
-        self.gp_eval_num += 1
-
-        if verbose:
-            print('[{0}/{1}]\teval_time={2:.2f} sec\tcurrent_{3}={4:.6f}\tmin_{3}={5:.6f}'.format(
-                self.gp_eval_num, self.gp_evals, eval_time,
-                model.metric, loss, self.best_loss))
-        return loss
-
-    def fit_eval(self, model, dtrain, dtest, params=None, n_estimators=None, custom_metric=None):
-        params = params or self.best_params or self.default_params
-        n_estimators = n_estimators or self.best_n_estimators or self.n_estimators
-        if params == None:
-            params = model.default_params
-        params = model.preprocess_params(params)
-        start_time = time.time()
-        _dtrain = model.convert_to_dataset(dtrain.X, dtrain.y, dtrain.cat_cols)
-        _dtest = model.convert_to_dataset(dtest.X, dtest.y, dtest.cat_cols)
-
-        bst, evals_result = model.fit(params, _dtrain, _dtest, n_estimators)
-        eval_time = time.time() - start_time
-
-        result = {'loss': evals_result[-1], 'bst': bst, 'n_estimators': n_estimators,
-                  'eval_time': eval_time, 'params': params.copy()}
-
-        if custom_metric is not None:
-            if type(custom_metric) is not dict:
-                raise TypeError("custom_metric argument should be dict")
-            prediction = model.predict(bst, _dtest, dtest.X)  # TODO: why 2 args?
-            for metric_name, metric_func in custom_metric.items():
-                score = metric_func(_dtest.get_label(), prediction, sample_weight=None)  # TODO weights
-                result[metric_name] = score
-
-        return result
+    def print_result(self, result, name='', extra_keys=None):
```

Comment:
if everything is the same, why bother redefining?

Reply:
ok